[14.0][IMP] delivery_schenker : submit booking

vnikolayev1 opened the pull request:
- Added field for schenker delivery carrier, where user can pick if they want to submit booking on schenker site. In case where it is turned off - user has to submit booking manually.

mt-software-de commented on delivery_schenker/models/delivery_carrier.py:
```diff
@@ -515,6 +522,11 @@ def _prepare_schenker_shipping(self, picking):
+                # Defines if booking shall be submitted. If false,
+                # the booking can be edited
+                # in the frontend and MUST be submitted manually.
+                "submitBooking": picking.carrier_id.is_schenker_submit_booking,
```
```suggestion
                "submitBooking": self.is_schenker_submit_booking,
```

mt-software-de commented on delivery_schenker/models/delivery_carrier.py:
```diff
@@ -179,6 +179,13 @@ class DeliveryCarrier(models.Model):
         help="If set, this contact will be sent as invoice address to Schenker."
         "\nIf Address ID is set, it will be part of it instead of the sender",
     )
+    is_schenker_submit_booking = fields.Boolean(
+        string="Submit Booking",
+        default=True,
+        help="Submit Booking On Schenker site."
+        "\nIf set, submits booking on site automatically."
+        "\nIf not set, user has to submit booking manually.",
+    )
```
Could you please change the name of the field to schenker_submit_booking
```suggestion
    schenker_submit_booking = fields.Boolean(
        string="Submit Booking",
        default=True,
        help="Submit Booking On Schenker site."
        "\nIf set, submits booking on site automatically."
        "\nIf not set, user has to submit booking manually.",
    )
```

mt-software-de commented on delivery_schenker/__manifest__.py:
```diff
@@ -4,7 +4,7 @@
     "name": "Delivery Schenker",
     "summary": "Delivery Carrier implementation for DB Schenker API",
-    "version": "14.0.1.2.0",
+    "version": "14.0.1.2.1",
```
Please remove this change. This should be automatically done by the merge bot.

mt-software-de commented on delivery_schenker/tests/common.py:
```diff
@@ -58,6 +58,12 @@ def setUpClass(cls):
             }
         )
         cls.sale, cls.picking = cls._create_sale_order(cls)
+        cls.picking.write(
+            {
+                "shipping_weight": 1,
+                "number_of_packages": 1,
+            }
+        )
```
What is the purpose of this change?

mt-software-de reviewed (changes requested):
Thanks for contributing

vnikolayev1 committed:
[IMP] delivery_schenker: configurable submit booking

- Added field for schenker delivery carrier, where user can pick if they want to submit booking on schenker site. In case where it is turned off - user has to submit booking manually.

jbaudoux reviewed (approved):
LGTM, not tested

jbaudoux commented:
@vnikolayev1 Can you rebase and repush as pre-commit failed

vnikolayev1 commented:
@jbaudoux sorry for delay, should be good now

mt-software-de commented:
> @jbaudoux sorry for delay, should be good now

Please rebase and not merge. To get rid of the second commit.

jbaudoux commented:
/ocabot merge minor
